pkg/sso/client.go

```diff
@@ -87,11 +91,18 @@ func (c *Client) GetAuthorizationURL(opts GetAuthorizationURLOptions) (*url.URL,
 	c.once.Do(c.init)
 
 	query := make(url.Values, 5)
-	query.Set("domain", opts.Domain)
 	query.Set("client_id", c.ProjectID)
 	query.Set("redirect_uri", c.RedirectURI)
 	query.Set("response_type", "code")
 
+	if opts.Domain == "" && opts.Provider == "" {
+		return nil, errors.New("Incomplete arguments. Need to specify either a 'domain' or 'provider'")
+	} else if opts.Domain == "" {
+		query.Set("provider", opts.Provider)
```

Comment:
is it possible to have `domain` and `provider` set at the same time?

if not we should do this check and throw an error.

Reply:
As you pointed on Slack, the API is doing a check on the provider first.
Doing a check on the provider first would improve code readability
```go
else if opts.Provider != "" {
    query.Set("provider", opts.Provider)
} else {
    query.Set("domain", opts.Domain)
}
```